### ISSUE TITLE
Allow python lists/arrays in KernelConnector

### ISSUE THREAD

**Issue by andrewgait** [SpiNNakerManchester/sPyNNaker#1323]
Currently the arrays passed in to a KernelConnector need to be converted into a numpy array by the user first (see https://github.com/SpiNNakerManchester/sPyNNaker/blob/e2c3d66ca2831e2e4ef239eca07d00cc88a17c35/spynnaker/pyNN/models/neural_projections/connectors/kernel_connector.py#L229).  This seems unnecessary so I suggest it gets fixed so that an array (of arrays) can be passed in as well.

**Comment by rowleya** [comment]
Yes, I think this was what was done some time ago; normal arrays should work I would guess (or they can be converted by the constructor if that is easier)
